zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/CommitProcessor.java

```diff
@@ -624,21 +626,19 @@ public void shutdown() {
 
         try {
             this.join(workerShutdownTimeoutMS);
+            if (this.isAlive()) {
+                LOG.warn("CommitProcessor does not shutdown gracefully after "
+                        + "waiting for {} ms, exit to avoid potential "
+                        + "inconsistency issue", workerShutdownTimeoutMS);
+                System.exit(ExitCode.SHUTDOWN_UNGRACEFULLY.getValue());
```

Comment:
Catch (InterruptedException) won't help for System.exit.

Maybe the best idea is to have a system property to disable the real 'System.exit' and add such property in tests (Maven surefire plugin config)